### PR TITLE
doc: nrfx: fix missing refhal and undocumented parameters

### PR DESCRIPTION
**Pull request by nika-nordic**
GENERATE_HTML must be set to avoid warnings about undocumented parameters. Additionally, alias for refhal function was missing.

Ref: NRFX-5192

### CODE REVIEW

File: doc/nrfx/nrfx.doxyfile.in
Comment (gmarull) [inline doc/nrfx/nrfx.doxyfile.in:1104]: does this really make some warnings disappear? is this a known doxygen bug?

Comment (nika-nordic) [reply doc/nrfx/nrfx.doxyfile.in:1104]: I have been looking for diff between Doxyfile from NCS and the one we have in nrfx. We have this symbol set to `YES` in nrfx that we don't observe such issues in there, so I have tried that in NCS context and it helped. I am not sure whether this is related to any specific bug

Comment (nika-nordic) [reply doc/nrfx/nrfx.doxyfile.in:1104]: @gmarull I have found someone stumbled upon similar issue here: https://stackoverflow.com/questions/37900027/doxygen-parameters-of-member-my-function-are-not-all-documented unfortunately without any specific conclusion

Comment (gmarull) [reply doc/nrfx/nrfx.doxyfile.in:1104]: Doxygen mysteries, let's go with it for now.

**Review by gmarull — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]
bit scary to change documentation build config after RC1.

Check the output is OK before merging.

**Comment by jfischer-no** [comment]
@gmarull, it requires +1 from @nrfconnect/ncs-doc-owners

**Review by b-gent — APPROVED** [review]

**Comment by b-gent** [comment]
Approved, please merge

**Pull request merged**
